File: commerce/src/main/java/com/epam/esm/purchase/models/PurchaseDTO.java
package com.epam.esm.purchase.models;

import com.epam.esm.purchasecertificate.model.PurchaseCertificate;
import com.fasterxml.jackson.annotation.JsonFormat;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Set;

import static com.epam.esm.utils.Constants.DATE_PATTERN;


public record PurchaseDTO(
        Long id,
        String description,
        BigDecimal cost,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = DATE_PATTERN)
        LocalDateTime createDate,
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = DATE_PATTERN)
        LocalDateTime lastUpdateDate,
        Long userId,
        Set<PurchaseCertificate> purchaseCertificates) {

}

File: notification/src/main/kotlin/com/epam/esm/utils/amqp/dto/PurchaseDTO.kt
package com.epam.esm.utils.amqp.dto

import java.math.BigDecimal
import java.time.LocalDateTime

data class PurchaseDTO(
    var id: Long = -1,
    var description: String? = "",
    var cost: BigDecimal = BigDecimal(-1),
    var createDate: LocalDateTime = LocalDateTime.MIN,
    var lastUpdateDate: LocalDateTime = LocalDateTime.MIN,
    var userId: Long = -1,
    var purchaseCertificates: Set<PurchaseCertificate> = emptySet()
)